openmetadata-ui/src/main/resources/ui/src/utils/FeedUtils.ts

```diff
@@ -95,3 +108,125 @@ export const getEntityFieldThreadCounts = (
 export const getThreadField = (value: string, separator = '/') => {
   return value.split(separator).slice(-2);
 };
+
+export async function suggestions(searchTerm: string, mentionChar: string) {
+  if (mentionChar === '@') {
+    let atValues = [];
+    if (!searchTerm) {
+      const data = await getInitialUsers();
+      const hits = data.data.hits.hits;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      atValues = hits.map((hit: any) => {
+        return {
+          id: hit._id,
+          value: `@${hit._source.display_name}`,
+          link: `${document.location.protocol}//${document.location.host}/${hit._source.entity_type}/${hit._source.name}`,
+        };
+      });
+    } else {
+      const data = await getUserSuggestions(searchTerm);
+      const hits = data.data.suggest['table-suggest'][0]['options'];
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      atValues = hits.map((hit: any) => {
+        return {
+          id: hit._id,
+          value: `@${hit._source.display_name}`,
+          link: `${document.location.protocol}//${document.location.host}/${hit._source.entity_type}/${hit._source.name}`,
+        };
+      });
+    }
+
+    return atValues;
+  } else {
+    let hashValues = [];
+    if (!searchTerm) {
+      const data = await getInitialEntity();
+      const hits = data.data.hits.hits;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      hashValues = hits.map((hit: any) => {
+        return {
+          id: hit._id,
+          value: `#${hit._source.name}`,
+          link: `${document.location.protocol}//${document.location.host}/${hit._source.entity_type}/${hit._source.fqdn}`,
+        };
+      });
+    } else {
+      const data = await getSuggestions(searchTerm);
+      const hits = data.data.suggest['table-suggest'][0]['options'];
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      hashValues = hits.map((hit: any) => {
+        return {
+          id: hit._id,
+          value: `#${hit._source.name}`,
+          link: `${document.location.protocol}//${document.location.host}/${hit._source.entity_type}/${hit._source.fqdn}`,
+        };
+      });
+    }
+
+    return hashValues;
+  }
+}
+
+export async function matcher(
+  searchTerm: string,
+  renderList: Function,
+  mentionChar: string
+) {
+  const matches = await suggestions(searchTerm, mentionChar);
+  renderList(matches, searchTerm);
+}
+
+const getMentionList = (message: string) => {
+  return message.match(mentionRegEx);
+};
+
+const getHahsTagList = (message: string) => {
```

Comment:
HashTag instead of HahsTag

Reply:
Thanks @mithmatt , for catching this typo.